pkg/cmd/roachtest/tests/kv.go

```diff
@@ -896,3 +897,130 @@ func registerKVRangeLookups(r registry.Registry) {
+			splits := ifLocal(c, " --splits=3", " --splits=20000")
+			c.Run(ctx, c.Node(workloadNode), "./cockroach workload init kv "+splits+" {pgurl:1}")
+
+			t.Status("starting kv workload thread to run for ", testDuration)
+			m := c.NewMonitor(ctx, c.Range(1, nodes))
+			m.Go(func(ctx context.Context) error {
+				testDurationStr := " --duration=" + testDuration.String()
+				concurrency := ifLocal(c, "  --concurrency=8", " --concurrency=64")
+				// Don't include the last node when starting the workload since it will
+				// stop in the middle, even with tolerate-errors set, it is still better
+				// not to use. Write enough data per value to make sure we create a
+				// large raft backlog.
+				c.Run(ctx, c.Node(workloadNode),
+					"./cockroach workload run kv --min-block-bytes=8192 --max-block-bytes=8192 --tolerate-errors --read-percent=50 "+
+						testDurationStr+concurrency+fmt.Sprintf(" {pgurl:1-%d}", nodes-1),
+				)
+				return nil
+			})
+
+			// Let some data be written to all nodes in the cluster.
+			t.Status("waiting ", duration, " to establish a base QPS")
+			time.Sleep(duration)
+			qpsInitial := measureQPS(ctx, t, db, 5*time.Second)
+			t.Status("initial (single node) qps: ", qpsInitial)
+
+			// Disable replicate queue on all nodes. This allows the test to reproduce
+			// the issue without a lot of fill beforehand. The system won't try and
+			// upreplicate these ranges somewhere else. We want to measure the impact
+			// of raft catchup, not snapshot movement.
+			setReplicateQueueEnabled := func(enabled bool) {
+				for n := 1; n <= nodes; n++ {
+					conn := c.Conn(ctx, t.L(), n)
+					defer conn.Close()
+					_, err := conn.ExecContext(ctx,
+						`SELECT crdb_internal.kv_set_queue_active('replicate', $1)`, enabled)
+					require.NoError(t, err)
+				}
+			}
+			setReplicateQueueEnabled(false)
+
+			// Gracefully shut down the last node to let it transfer leases cleanly.
+			// Wait enough time to let it fall behind on Raft. Since there are a lot
+			// of ranges, only a small number will be upreplicated during this time.
+			gracefulOpts := option.DefaultStopOpts()
+			gracefulOpts.RoachprodOpts.Sig = 15 // SIGTERM
+			gracefulOpts.RoachprodOpts.Wait = true
+			c.Stop(ctx, t.L(), gracefulOpts, c.Node(nodes))
+			t.Status("waiting ", duration, " after stopping node to allow the node to fall behind")
+			time.Sleep(duration)
+
+			// Start the node again. It will attempt to catch up and go into an IO
+			// Overload scenario. Re-enable the replicate queue now so that leases
+			// begin to transfer.
+			t.Status("restarting stopped node and the replicate queue")
+			c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.Node(nodes))
+			setReplicateQueueEnabled(true)
+			t.Status("waiting ", duration, " for the workload to finish and measuring the impact of the outage")
+
+			// Wait for IO overload and enough leases to be transferred back.
```

Comment:
Perhaps include a stock grafana dashboard so you also get graphs + a prometheus dump for posterity.